Godeps/_workspace/src/github.com/facebookgo/counting/reader.go

```diff
@@ -0,0 +1,30 @@
+package counting
+
+import (
+	"io"
+)
+
+// Reader wraps an existing io.Reader and keeps track of how much has been read
+// from it.
+type Reader struct {
+	reader io.Reader
+	count  int
+}
+
+// Read some bytes into b and return the number of bytes read and error if any.
+func (r *Reader) Read(b []byte) (int, error) {
+	n, err := r.reader.Read(b)
+	r.count += n
```

Comment:
Lets use the `sync/atomic` package here to make things threadsafe:
http://golang.org/pkg/sync/atomic/#AddUint64

Should be a lot cheaper than locks, which is important for something that is going to be hit so much